src/optimizer/join_order/query_graph_manager.cpp

```diff
@@ -368,6 +369,25 @@ unique_ptr<LogicalOperator> QueryGraphManager::LeftRightOptimizations(unique_ptr
 							cond.comparison = FlipComparisonExpression(cond.comparison);
 						}
 					}
+				} else if (join.join_type == JoinType::LEFT_SEMI || join.join_type == JoinType::LEFT_ANTI) {
```

Comment:
Would it be possible to create separate cases for `LEFT_SEMI` and `LEFT_ANTI` to improve readability? I know there's a lot of duplicated code, but we can de-duplicate most of it by a function like this:
```c++
void TryFlipSides(LogicalComparisonJoin &join, JoinType inverse, bool flip_comparison_expressions) {
    auto lhs_cardinality = join.children[0]->EstimateCardinality(context);
    auto rhs_cardinality = join.children[1]->EstimateCardinality(context);
    if (rhs_cardinality < lhs_cardinality * 2) {
        return;
    }
    join.join_type = inverse;
    std::swap(join.children[0], join.children[1]);
    if (!flip_comparison_expressions) {
        return;
    }
    for (auto &cond : join.conditions) {
        std::swap(cond.left, cond.right);
        cond.comparison = FlipComparisonExpression(cond.comparison);
    }
}
```